pkg/generate/code/common.go

```diff
@@ -162,12 +156,12 @@ func FindPrimaryIdentifierFieldNames(
 		// Require override if still can't find any identifiers
 		if shapeField == "" {
 			panic("Could not find primary identifier for " + r.Names.Original +
-				". Set `primary_identifier_field_name` for the " + op.Name +
-				" operation in the generator config.")
+				". Set `is_primary_key` for the primary field in the " +
+				r.Names.Camel + " resource.")
 		}
 	}
 
-	if r.IsPrimaryARNField(shapeField) || shapeField == PrimaryIdentifierARNOverride {
+	if r.IsPrimaryARNField(shapeField) {
```

Comment:
ApplicationAutoscaling does not utilize this, so I have not verified this snippet, looks ok though.

Reply:
The SageMaker controller uses it (for ModelPackage). We have unit tests that cover it.